<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">
    <title>Protein Sequence Coverage Map</title>

    <link type="text/css" rel="stylesheet" href="../static/css/materialize.min.css" media="screen,projection" />
    <link type="text/css" rel="stylesheet" href="../static/css/fonts.css" media="screen,projection">
    <link type="text/css" rel="stylesheet" href="../static/css/style.css" media="screen,projection">
    <link type="text/css" rel="stylesheet" href="./style.css" media="screen,projection">
    <script src="https://kit.fontawesome.com/cb26641bd1.js"></script>
    <link rel="shortcut icon" href="../static/img/favicon.ico" type="image/x-icon">
</head>

<body>
    <nav class="au-color">
        <div class="nav-wrapper container">
            <a href="../" class="brand-logo">
                <img src="../static/img/auvis_logo_large_white.png" height="45px" alt="">
            </a>
            <ul class="right hide-on-med-and-down">
                <li><a href="../publications/">Publications</a></li>
                <li><a class="current" href="../software/">Software</a></li>
                <li><a href="../teaching_materials/">Teaching Material</a></li>
                <li><a href="../student_projects/">Student Projects</a></li>
                <li><a href="../workshops/">Workshops</a></li>
                <li><a href="../team/">Team</a></li>
            </ul>
        </div>
    </nav>

    <article class="container">
        <section class="row" id="map-like">
            <h2>Protein Sequence Coverage Map</h2>
            <p class="leader">Protein sequence coverage maps are visualizations used in proteomics and peptidomics to
                show the distribution of peptides across their parent protein. This software was developed in
                collaboration with Aarhus University’s Department of Food Science. Further details on the software and
                its use are provided below. To reference the software, please cite the accompanied review article on
                bioactive milk peptides. </p>
        </section>

        <section class="row">
            <div class="col s12 m7 abstract">
                <h3>Authors</h3>
                <div id="author-list">
                    <a href="https://pure.au.dk/portal/en/persons/harith-rathish(abd9352b-490f-4118-b668-30497074c27e).html"
                        target="_blank" class="author"><span>Harith Rathish</span></a>,
                    <a href="https://www.linkedin.com/in/s%C3%B8ren-drud-heydary-nielsen-76829225/?originalSubdomain=dk"
                        target="_blank" class="author"><span>Søren Drud-Heydary Nielsen</span></a>,
                    <a href="https://hjschulz.net/" class="author"><span>Hans-Jörg Schulz</span></a>
                </div>

                <figure class="teaser" style="color:#777;border: 1px solid #ccc">
                    <a href="https://youtu.be/TnAurhwSrhs" target="_blank">
                        <img src="../static/img/protein-seq-map-annotated.png"
                            alt="Cover image of Protein Sequence Map." />
                    </a>
                    <figcaption>Cover image of Protein Sequence Map. Details of the labels are given in the description.
                        <b>Click this image to watch the demo video.</b>
                    </figcaption>
                </figure>

                <h3>Overview</h3>
                The protein sequence coverage map is widely used (<a
                    href="https://www.researchgate.net/figure/Sequence-coverage-map-of-full-length-EBP50-Bars-indicate-the-identified-peptic-peptides_fig2_273780845"
                    style="color: blue;font-weight: bolder;">1</a>, <a
                    href="https://www.researchgate.net/figure/Coverage-map-of-the-peptide-map-obtained-by-the-characterization-stage-of-MAM-analysis_fig4_345028794"
                    style="color: blue;font-weight: bolder;">2</a>, <a
                    href="https://www.researchgate.net/figure/Protein-sequence-coverage-by-mass-spectrometry-of-the-serpin-purified-from-the-hemolymph_fig5_259723603"
                    style="color: blue;font-weight: bolder;">3</a>) to provide an overview of the peptides associated to
                a protein. This implementation is targeted for <b>use cases where horizontal space is limited, e.g., in
                    a paper.</b>

                <h3>Description</h3>
                <p>The visualizations show a horizontal line <b>(1)</b> for each peptide, with a different color for
                    different bioactive functions <b>(2)</b>. The purpose of this plot is to provide an overview of all
                    the peptides that were associated with the protein. The lines that wrap around the edges of the
                    image are indicated with an arrow mark <b>(3)</b>.

                    For each amino acid "X", the longest peptide that starts with "X" is shown on top. From top to down,
                    the length of the peptides decrease. </p>
                <p> Currently we support exporting the plot as an SVG image <b>(4) </b>, which contains the plot
                    alone without the legend or the controls.</p>


                <h3>Interaction</h3>

                <p><b>Choose a protein (5) </b>: Users can choose a protein out of the list of unique values for
                    <b>"Entry"</b> from the protein sequences dataset. The sequence will be built and peptides will be
                    stacked based on this choice.
                </p>
                <p><b>Length of signal peptide (6) </b>: This indicates the number of amino acids to hide from the
                    beginning of the
                    sequence. </p>
                <p><b>Max axis length (7) </b>: This indicates the maximum number of amino acids shown per axis. If it's
                    set to 0, then the number is calculated based on available screen width. This lets users control the
                    width according to their horizontal space limits.</p>


                <h3>Design Choices</h3>

                <h4> Color Scheme</h4>

                <p>The plot uses a categorical color scheme prescribed in Figure 4 of Qualititative Color
                    Schemes in <a href="https://personal.sron.nl/~pault/" target="_blank" style="color:blue">Paul Tol's
                        notes</a>. We chose this scheme as it is developed with the help of mathematical descriptions of
                    colour differences and the two main types of colour-blind vision. The only change we've made is
                    darkening the grey color from #DDDDDD to #BBBBBB, due to printability reasons. </p>
                <p>As of now, the bioactive functions map to the same colors as shown in the
                    figure, regardless of their order in any input dataset. The color codes for each bioactive function
                    is shown in the table below. </p>
                <table>
                    <th>
                    <td><b>Bioactive Function</b></td>
                    <td><b>Color Hexcode</b></td>
                    </th>
                    <tr>
                        <td></td>
                        <td>ACE-inhibitory</td>
                        <td style="color: #CC6677;"> <b>#CC6677</b>
                        </td>
                    </tr>

                    <tr>
                        <td></td>
                        <td>Antimicrobial</td>
                        <td style="color: #DDCC77;"> <b>#DDCC77</b>
                        </td>
                    </tr>
                    <tr>
                        <td></td>
                        <td>Antioxidant</td>
                        <td style="color: #117733;"> <b>#117733</b>
                        </td>
                    </tr>
                    <tr>
                        <td></td>
                        <td>DPP-IV Inhibitor</td>
                        <td style="color: #88CCEE;"> <b>#88CCEE</b>
                        </td>
                    </tr>
                    <tr>
                        <td></td>
                        <td>Opioid</td>
                        <td style="color: #999933;"> <b>#999933</b>
                        </td>
                    </tr>
                    <tr>
                        <td></td>
                        <td>immunomodulatory</td>
                        <td style="color: #44AA99;"> <b>#44AA99</b>
                        </td>
                    </tr>
                    <tr>
                        <td></td>
                        <td>Anticancer</td>
                        <td style="color: #AA4499;"> <b>#AA4499</b>
                        </td>
                    </tr>
                    <tr>
                        <td></td>
                        <td>Others</td>
                        <td style="color: #BBBBBB;"> <b>#BBBBBB</b>
                        </td>
                    </tr>
                </table>
                </p>
                <br />
                <h4>Structure of input data</h4>
                There are two datasets to be uploaded in <b>.csv</b> format onto fields <b>(8)</b> and <b>(9)</b>, the
                structure of which are as follows:
                <p> <b>Protein Sequences</b> <br />
                    Should contain two columns, one for the protein ID ("Entry") and one for the protein sequence
                    ("Sequence"). The protein ID should be unique, and should match an entry in the <b>
                        UniProt</b> database. The sequence string should have no spaces or line breaks. An example is
                    shown below:
                <table>
                    <caption>Example for protein sequences </caption>
                    <th>
                    <td><b>Entry</b></td>
                    <td><b>Sequence</b></td>
                    </th>
                    <tr>
                        <td></td>
                        <td>P02663</td>
                        <td>MKFFIFTCL...
                        </td>
                    </tr>

                    <tr>
                        <td></td>
                        <td>P02662</td>
                        <td>MKLLILTCL...
                        </td>
                    </tr>
                    <tr>
                        <td>

                        </td>
                        <td>...</td>
                        <td>...</td>
                    </tr>
                </table>
                </p>
                <br />
                <p> <b>Peptides</b> <br />
                    Should contain three columns, one for the protein ID ("proteinID"), one for the peptide sequence
                    ("Peptide") and one for the bioactive function of the peptide ("function"). The protein ID should
                    match one and only one entry in the protein sequences dataset.
                <table>
                    <caption>Example for peptides </caption>
                    <th>
                    <td><b>proteinID</b></td>
                    <td><b>peptide</b></td>
                    <td><b>function</b></td>
                    </th>
                    <tr>
                        <td></td>
                        <td>P02663</td>
                        <td>TKVIPYVRYL
                        </td>
                        <td>Antimicrobial</td>
                    </tr>

                    <tr>
                        <td></td>
                        <td>P02662</td>
                        <td>FFVAP
                        </td>
                        <td>
                            ACE-inhibitory
                        </td>
                    </tr>
                    <tr>
                        <td>

                        </td>
                        <td>...</td>
                        <td>...</td>
                        <td>...</td>
                    </tr>
                </table>
                </p>
                <br />

            </div>

            <div class="col s12 m3 right resources">
                <h3>Resources</h3>

                <!-- <h4> <a href=""><i class="icon fas fa-file-pdf"></i> Paper</a></h4> -->
                <h4> <a href="https://doi.org/10.1080/10408398.2023.2240396" target="_blank"><i
                            class="icon fas fa-globe"></i>
                        Publisher</a></h4>
                <h4><a href="https://doi.org/10.1080/10408398.2023.2240396" target="_blank"><i
                            class="icon fas fa-file-pdf"></i> Paper</a>
                </h4>
                <h4><a href="../publications/bib/Harith2023_peptides.bib" target="_blank"><i
                            class="icon fas fa-quote-right"></i> BibTeX
                        Citation</a></h4>
                <br />
                <h4><a href="https://harith1996.github.io/peptide_bioactive_viz" target="_blank"><i
                            class="icon fas fa-external-link-alt"></i> Try it online!</a></h4>
                <h4><a href="https://youtu.be/TnAurhwSrhs" target="_blank"><i class="icon fas fa-video"></i> Demo
                        video</a></h4>
                <h4><a href="../static/resources/psm_example_dataset.zip"><i class="icon fas fa-database"></i> Example dataset</a>
                </h4>
                <h4><a href="https://github.com/harith1996/peptide_bioactive_viz/releases/tag/v1"><i
                            class="icon fas fa-download"></i>
                        Download</a></h4>
                <h4 class="highlight"><a href="https://github.com/harith1996/peptide_bioactive_viz" target="_blank"><i
                            class="icon fab fa-github"></i> Code on Github</a></h4>
            </div>
        </section>

        <section class="row">
            <div class="col s12 m7 abstract">
                <h3>Citation in BibTeX</h3>

                <p>To cite this article, we encourage you to use the following bibtex entry in your citation manager:
                </p>

                <pre>
@article{doi:10.1080/10408398.2023.2240396,
author = {Søren Drud-Heydary Nielsen and Ningjian Liang and Harith Rathish and Bum Jin Kim and Jiraporn Lueangsakulthai and Jeewon Koh and Yunyao Qu and Hans-Jörg Schulz and David C. Dallas},
title = {Bioactive milk peptides: an updated comprehensive overview and database},
journal = {Critical Reviews in Food Science and Nutrition},
volume = {0},
number = {0},
pages = {1-20},
year  = {2023},
publisher = {Taylor & Francis},
doi = {10.1080/10408398.2023.2240396},
    note ={PMID: 37504497},
URL = { 
        https://doi.org/10.1080/10408398.2023.2240396
},
eprint = { 
        https://doi.org/10.1080/10408398.2023.2240396
}
}
        </pre>
            </div>
        </section>

        <section class="empty"></section>
    </article>

    <footer class="page-footer au-color darken-1">
        <div class="container">
            <div class="row">
                <div class="col l6 s12">
                    <h5 class="white-text">Aarhus University Visualization Repository</h5>
                    <p class="grey-text text-lighten-4">
                    </p>
                </div>
                <div class="col l4 offset-l2 s12">
                    <ul>
                        <li><a class="grey-text text-lighten-3" href="https://cs.au.dk/">Department of Computer Science,
                                Aarhus University</a></li>
                        <li><a class="grey-text text-lighten-3"
                                href="https://cs.au.dk/research/ubiquitous-computing-and-interaction/">Research Group
                                for Ubiquitous Computing and Interaction</a></li>
                    </ul>
                </div>
            </div>
        </div>
        <div class="footer-copyright">
            <!-- <div class="container">
        © 2020 Copyright Department of Computer Science, Aarhus University
      <a class="grey-text text-lighten-4 right" href="#">back to top</a>
      </div> -->
        </div>
    </footer>

    <script type="text/javascript" src="../static/js/materialize.min.js"></script>
</body>

</html>